### DISCUSSION TITLE
How can I check if a file already exists in the indexDB?

### DISCUSSION THREAD

**Discussion by eugene-f02** [jupyterlite/jupyterlite#861] [Q&A]
Hi! 
I'm working on creating a jupyterlite front-end extension where the plugin information contains the following:
``` 
const my_extension: JupyterFrontEndPlugin<void> = {
  id: "my_extension",
  autoStart: true,
  requires: [IDocumentManager, INotebookTracker, INotebookWidgetFactory]

...
}
```

and inside the `activate` function, I'm using an instance of IDocumentManager (which I named `docManager`) and I'm calling
```
let npPanel = docManager.createNew(filename, 'notebook', { name:  "pyhton"}) as NotebookPanel
nbPanel.model!.fromString(content);
```
in order to create a new notebook and populate it with some specific content. However, If the file already exists in the indexDB, I don't want to recreate this file and/or change its content.

I would like to know if there is a way for me to check if a certain file is already in the index DB and, preferably, extract the metadata about the file from indexDB inside my `activate` function.

I tried
```
await docManager.services.contents.get(filename)
```
but it throws an error
```
SyntaxError: Unexpected end of input (at serverconnection.js:102:45)
    at t.create (serverconnection.js:102:45)
    at _.get (index.js:509:66)
    at async index.js:73:1
``` 

Is there any other approach that could work? Thanks in advance
@jtpio

**Comment by jtpio** [comment]
@eugene-f02 I would have expected the `docManager` to work in this case. Another way would be to use the `app.serviceManager`.

Anyway it's indeed better to use the proper JupyterLab API and services like you mention above, instead of trying to read from IndexedDB directly. Normally this extension should work in both JupyterLab and JupyterLite without modifications.

> However, If the file already exists in the indexDB, I don't want to recreate this file and/or change its content.

Would that behavior be the same in case the file is served statically (available on the server)? That would be simpler if yes.
